src/SWP/UpdaterBundle/Manager/AbstractManager.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace SWP\UpdaterBundle\Manager;
+
+use SWP\UpdaterBundle\Client\ClientInterface;
+use vierbergenlars\SemVer\version;
+
+/**
+ * Abstract update manager.
+ *
+ * @author Rafał Muszyński <rafal.muszynski@sourcefabric.org>
+ */
+abstract class AbstractManager implements ManagerInterface
+{
+    /**
+     * Current app version.
+     *
+     * @var string
+     */
+    protected $currentVersion = '0.0.0';
+
+    /**
+     * List of available updates.
+     *
+     * @var array
+     */
+    protected $updates = array();
+
+    /**
+     * The latest app version.
+     *
+     * @var string
+     */
+    protected $latestVersion = '';
+
+    /**
+     * @var ClientInterface
+     */
+    protected $client;
+
+    /**
+     * Construct.
+     *
+     * @param ClientInterface $client Update server client
+     */
+    public function __construct(ClientInterface $client)
+    {
+        $this->client = $client;
+        // TODO
+        // inject here Setting service to get the current app version.
```

Comment:
I think that would be better to keep version in class file, not in storage (database).